bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/metrics/MetricsApiTracerAdapter.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.wrappers.veneer.metrics;
+
+import com.google.api.gax.tracing.ApiTracer;
+import com.google.cloud.bigtable.metrics.RpcMetrics;
+import com.google.cloud.bigtable.metrics.Timer.Context;
+import com.google.common.base.Stopwatch;
+import org.threeten.bp.Duration;
+
+public class MetricsApiTracerAdapter implements ApiTracer {
+
+  private final RpcMetrics rpcMetrics;
+  private Context operationTimer;
+  private Context rpcTimer;
+
+  private int attemptCount = 0;
+  private Stopwatch attemptTimer;
+  private long attemptResponseCount = 0;
+  private volatile RetryStatus call;
+
+  public MetricsApiTracerAdapter(RpcMetrics rpcMetrics) {
+    this.rpcMetrics = rpcMetrics;
+    operationTimer = rpcMetrics.timeOperation();
+  }
+
+  @Override
+  public Scope inScope() {
+    return new Scope() {
+      @Override
+      public void close() {}
+    };
+  }
+
+  @Override
+  public void operationSucceeded() {
+    operationTimer.close();
+  }
+
+  @Override
+  public void operationCancelled() {}
+
+  @Override
+  public void operationFailed(Throwable error) {
+    operationFailed(error, call);
+  }
+
+  public void operationFailed(Throwable error, RetryStatus retryStatus) {
```

Comment:
this seems to be called only by `operationFailed(Throwable error)`. Can this be inlined?